flo-workflow/src/main/java/com/spotify/flo/context/InstrumentedContext.java

```diff
@@ -144,6 +158,16 @@ public static EvalContext composeWith(EvalContext baseContext, Listener listener
 
   @Override
   public TaskOperator.Listener listener() {
-    return super.listener().composeWith(listener::meta);
+    return super.listener().composeWith(new TaskOperator.Listener() {
```

Comment:
Difficult to put into words.
This will never be called from `EvalContext`. InstrumentedContext is a `ForwardingEvalContext`, so it forwards the evaluation of a task to another context which is composed within (rather than forwarding to the `super` implementation). According to https://github.com/spotify/flo/blob/master/flo-runner/src/main/java/com/spotify/flo/context/FloRunner.java#L168 that's the base `EvalContext` which is either `SyncContext` or `AsyncContext`. So it forwards the evaluation to another context instance which doesn't contain this listener. Therefore the listener here https://github.com/spotify/flo/blob/master/flo-workflow/src/main/java/com/spotify/flo/EvalContext.java#L92-L103 is simply NOP https://github.com/spotify/flo/blob/master/flo-workflow/src/main/java/com/spotify/flo/EvalContext.java#L180, and we end up calling meta on NOP :/

Reply:
Yeah, I guess [this line](https://github.com/spotify/flo/blob/master/flo-workflow/src/main/java/com/spotify/flo/EvalContext.java#L92) should be changed to:

```java
final Listener listener = context.listener();
```

And we should have some tests to cover this. Good catch!

Reply:
👍